app/services/auth_token_verifier.rb

```diff
@@ -19,7 +20,7 @@ def verify
   end
 
   def authentication
-    @authentication ||= MemberAuthentication.find_by(token: @token)
+    @authentication ||= @member.authentication
```

Comment:
If you want a more objective reason for passing member, this check should be faster since the member needs to be fetched in any case.